File: src/main/java/com/mbayou/kick4k/chat/ChatClient.java
package com.mbayou.kick4k.chat;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.mbayou.kick4k.KickConfiguration;
import com.mbayou.kick4k.api.ApiClient;
import com.mbayou.kick4k.authorization.AuthorizationClient;

import java.net.http.HttpClient;

public class ChatClient extends ApiClient {
    public ChatClient(HttpClient httpClient, ObjectMapper mapper, KickConfiguration configuration, AuthorizationClient authorization) {
        super(httpClient, mapper, configuration, authorization);
    }

    public PostChatMessageResponse postChatMessage(PostChatMessageRequest request) {
        return this.post(this.configuration.getChat())
                .body(request)
                .send(new TypeReference<>() {});
    }
}

File: src/main/kotlin/com/mbayou/kick4k/chat/ChatClient.kt
package com.mbayou.kick4k.chat

import com.fasterxml.jackson.databind.ObjectMapper
import com.mbayou.kick4k.KickConfiguration
import com.mbayou.kick4k.api.ApiClient
import java.net.http.HttpClient

class ChatClient(
    httpClient: HttpClient,
    mapper: ObjectMapper,
    configuration: KickConfiguration,
): ApiClient(httpClient, mapper, configuration) {

    fun postChatMessage(accessToken: String, request: PostChatMessageRequest): PostChatMessageResponse? {
        return post(configuration.chat)
            .body(request)
            .withAccessToken(accessToken)
            .send(responseType<PostChatMessageResponse>())
    }
}
